### PR TITLE
EuroSAT100: add new dataset/datamodule

### PR DESCRIPTION
**Pull request by adamjstewart**
#### What?

This is a subset of the EuroSAT MS dataset containing only 100 images. Yes, you heard that right, 100 images. This dataset maintains the 60-20-20 train-val-test split from https://arxiv.org/abs/1911.06721. There are 10 images (6 train, 2 val, 2 test) for all 10 classes, for a total of 100 images.

#### But why?

Our tutorials download a lot of data. So much in fact that we crash the meager 14 GB SSD in our [GitHub Actions runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), causing our notebook tests to fail for the last year or so. Between NAIP, Chesapeake, Sentinel, EuroSAT, and TropicalCyclone, we're talking several minutes of downloads. The TropicalCyclone download doesn't even work unless you sign up for an API key (#1074). A quick comparison:

| Dataset | Compressed | Uncompressed | Download Time |
| --- | --- | --- | --- |
| EuroSAT | 1.9G | 4.7G | 3m 7s |
| EuroSAT100 | 7.4M | 18M | 1s |

Yes, 1 second, down to the millisecond:
```console
$ time wget https://huggingface.co/datasets/torchgeo/eurosat/resolve/main/EuroSAT100.zip
...
real	0m1.000s
user	0m0.055s
sys	0m0.110s
```
Our tutorials should not take several minutes just on data prep, nor should they fill up your hard drive, nor should they require an API key.

#### Why EuroSAT?

EuroSAT was chosen for the following reasons:

* Many of our existing tutorials already use EuroSAT, making them easy to port
* It contains Sentinel-2 images, which our tutorials rely on to show off things like NDVI or pre-trained models
* It is distributed under a [permissive license](https://github.com/phelber/EuroSAT/issues/10#issuecomment-1409891796)
* Images are georeferenced, meaning we could add a GeoDataset version of this someday

#### The milestone says 0.4.1 but versionadded says 0.5...

Technically, new features shouldn't be added in patch releases. But this may fix our chronically failing notebook tests. So I would like to add this to the next patch release for the sake of testing, but we can just pretend that it was actually added in 0.5.

#### Hey, you forgot to update the docs!

I'll open a separate PR to add this to the docs that we can save for the 0.5 release.

### CODE REVIEW

File: torchgeo/datamodules/eurosat.py [hunk 1]
```diff
@@ -67,3 +67,59 @@ def __init__(
                 :class:`~torchgeo.datasets.EuroSAT`.
         """
         super().__init__(EuroSAT, batch_size, num_workers, **kwargs)
+
+
+class EuroSAT100DataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the EuroSAT100 dataset.
+
```
Comment (calebrob6) [inline torchgeo/datamodules/eurosat.py:74]: Can you add the "Intended for tutorials and demonstrations, not for benchmarking." disclaimer here too

**Comment by calebrob6** [comment]
Could not reproduce the 1 second download, it took my laptop 1.971 seconds.

File: torchgeo/datamodules/eurosat.py (hunk 1 above)
Comment (adamjstewart) [reply torchgeo/datamodules/eurosat.py:77]: Done.

**Comment by adamjstewart** [comment]
I tried the download several times and it's always between 0.8s and 1.8s on my laptop. Either way, drastic improvement over the previous 3m download for EuroSAT and 10+ min download for Cyclone.

**Review by calebrob6 — APPROVED** [review]

**Comment by calebrob6** [comment]
That's more like it! We emphasize test metric distributions in torchgeo!
